docs/source-control/git-guidance/README.md

```diff
@@ -232,6 +232,74 @@ Then you can use the reflog reference (`HEAD@{}`) to reset to a specific commit
 git reset HEAD@{2}
 ```
 
+## Commit Best Practices
```

Comment:
This section has bothered me for a while, and I am wondering if we can maybe re-phrase it a bit.

To me, a commit is a ```git commit -m "some message"``` and I personally think we should encourage people to commit extremely often and with fairly brief commit messages - ex. "implement GET /todos endpoint"

The guidance given here, is, in my opinion more geared towards PR descriptions, where one PR may contain 40 commits.

Do you agree? And if so, should we simply move this out all together, and add the relevant parts of this content to the guidance for code reviews.

Reply:
PS! I know this was part of a re-factor, but since this PR is a mix of re-factoring and functional code changes in one, it makes it fair game to comment :)

Reply:
I reduced the content a little more and only added the most essential "rules" that the git command expects.

Do we want to keep the sections Commit Best Practices and Commit Message? They are linked in other documents so might be good to keep them. However, we might want to clean them up a little further.

In particular, the following could be compressed into one:
```text
A good commit message should answer these questions:

* Why is it necessary? It may fix a bug, add a feature, improve performance, or just be a change for the sake of correctness
* How does it address the issue? For short, obvious changes, this can be omitted
* What effects does this change have? In addition to the obvious ones, this may include benchmarks, side effects etc.
* What limitations does the current code have?

Consider this when writing your commit message:

* If it seems difficult to summarize your commit, it may be because it includes more than one logical change or bug fix. If so, it is better to split it into separate commits with `git add -p`
* Don't assume the reviewer understands the original problem. It should be possible to review a change request without reading the contents of the original bug/task.

```

Reply:
If we remove them we can just remove the links from the other sections
I'll approve the PR and leave it up to you how you want to finish it up